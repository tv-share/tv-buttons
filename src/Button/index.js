import "./button.*";
import Button from "./Button";

export default Button;