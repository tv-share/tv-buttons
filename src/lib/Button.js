import React from "react";
import PropTypes from "prop-types";

/**
* Renders the Button Component
* @param  {String}    	className is a string that will be attached to the component's className (optional)
* @param  {Boolean}     disabled specifies if the button should be disabled
* @param  {String}    	label defines a label for the button
* @param  {Function}    onClick defines a function to be called whenever the button is pressed
* @param  {String}    	type defines they type of the button (button, reset or submit)
* @param  {String}    	value specifies an initial value for the button
* @return {JSX}         Button Component
*/
const Button = ({ className, disabled, label, onClick, type = "button", value, events = {} }) => {
	return (
        <button
            className={`tv-button ${className ? className : ""}`}
            disabled={disabled}
            onClick={onClick}
            type={type}
            value={value}
            {...events}>
            {label || ""}
		</button>
	);
};

Button.propTypes = {
    label: PropTypes.oneOfType([PropTypes.string, PropTypes.node]).isRequired,
    className: PropTypes.string,
    onClick: PropTypes.func,
    disabled: PropTypes.bool,
    type: PropTypes.string,
    value: PropTypes.string,
    events: PropTypes.object
};

export default Button;