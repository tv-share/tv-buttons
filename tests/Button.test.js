import React from 'react';
import renderer from 'react-test-renderer';

import Button from "../src/Button/Button";

const fn = () => { };

it('renders correctely', () => {
	const component = renderer.create(
		<Button label="Exemplo" onClick={() => console.log("Click")} />
	);

	expect(component.toJSON()).toMatchSnapshot();
});
