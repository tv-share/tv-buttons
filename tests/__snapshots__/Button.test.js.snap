// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctely 1`] = `
<button
  className="button "
  disabled={undefined}
  onClick={[Function]}
  type={undefined}
  value={undefined}
>
  Exemplo
</button>
`;
